api/envoy/data/accesslog/v3/accesslog.proto

```diff
@@ -236,6 +256,14 @@ message AccessLogCommon {
   // For HTTP: Total number of bytes received from the upstream by the http stream.
   // For TCP: Total number of bytes sent to the upstream by the tcp proxy.
   uint64 upstream_wire_bytes_received = 32;
+
+  // The type of the access log, which indicates when the log was recorded.
+  // See :ref:`ACCESS_LOG_TYPE <config_access_log_format_access_log_type>` for the available values.
+  // In case the access log was recorded by a flow which does not correspond to one of the supported
+  // values, then the default value will be ``NotSet``.
+  // For more information about how access log behaves and when it is being recorded,
+  // please refer to :ref:`access logging <arch_overview_access_logs>`.
+  AccessLogType access_log_type = 33;
```

Comment:
That's a good point, although as far as I understand, I wouldn't expect the log collector to drop them, because Protobuf should be forward compatible (unless explicitly drops unknown values). However, the case you describe can still happen. Would it make sense to change this to string instead of int32 then? Seems more approaching to methen an integer value. wdyt?

Reply:
Changed it to string type, please lmk if this makes sense

Reply:
cc @alaiacano cc @yanavlasov to see if uint32 or string should be used here to add the extendability.

Reply:
@yanavlasov giving this a second thought, I'm trying to understand what uint32 solves that enum doesn't. As far as I can see, the problem you're describing is a case where the collector service is (as example use case) handling the incoming log message and has a 'switch-case' over the values of the enum. In that case, you are correct that the handler side would probably get to the default case, as the new value is unknown. But isn't this the same with the integer? Assuming I'm adding a new value that's sent as an integer, the handler side still has a switch case to handle those mapped integers and would still be unfamiliar with the meaning of the new value, so the outcome is the same.

Looking at the problem more pragmatically, it feels to me that the majority of use cases in terms of those log collector services are just printing the log or sending it to some other dashboarding service to be able to view the logs and filter them for debugging/live-site purposes, so there wouldn't be some special handling for the value of the enum. For use cases where there is a special handling for the enum, the handler side should be aware that the enum values may be extended in the future (maybe we could also add a comment about it here), and this is a known issue, relevant not only to Protobuf.